api/types/container/hostconfig.go

```diff
@@ -79,7 +79,7 @@ type IpcMode string
 const (
 	IPCModeNone      IpcMode = "none"
 	IPCModeHost      IpcMode = "host"
-	IPCModeContainer IpcMode = "container"
+	IPCModeContainer IpcMode = "container" //nolint:unused // was used to match prefix, but we could consider removing, as it's not a literal value
```

Comment:
How come the nolint annotation is necessary? IIRC the linter always considers exported identifiers to be used.

As for what to do, my first instinct would be to (in a new major version of the module) change it to a convenience function which clients can use to build a mode string.
```go
func IPCModeContainer(idOrName string) IpcMode
```

Reply:
Hm.. let me double check, perhaps it's not needed no

And, yes, it's a bit unfortunate that UX leaked into the API in these cases. I don't think we ever should've had these formats here, as they're effectively a 1:1 copy of what's passed on the command-line 😞